expression/builtin_json.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2013 The ql Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSES/QL-LICENSE file.
+
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/util/types"
+)
+
+var (
+	_ functionClass = &jsonExtractFunctionClass{}
+)
+
+type jsonExtractFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *jsonExtractFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	return &builtinJsonExtractSig{newBaseBuiltinFunc(args, ctx)}, errors.Trace(c.verifyArgs(args))
+}
+
+type builtinJsonExtractSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinJsonExtractSig) eval(row []types.Datum) (d types.Datum, err error) {
```

Comment:
Add typeinferer for json built-in funcs is also needed to add in plan/typeinferer.go.